include/oneapi/tbb/collaborative_once_flag.h

```diff
@@ -61,7 +61,7 @@ class run_once {
                 cancellation_status = my_context.is_group_execution_cancelled();
             });
         };
-        my_arena.execute([&f]() { return tbb::this_task_arena::isolate(f); });
+        my_arena.execute(f);
```

Comment:
This change is needed to make the `collaborative_once_flag` actually run in parallel. I think it is correct because the first thread gets trapped inside the call once, so it cant deadlock, but then the other threads are free to do what they want. Does this seem right to you?

Reply:
Surely, it should not be a new isolated region as it was before. Should be it the same isolated region that created by the primary thread or no isolation is Ok here? There are pros and cons for the both options, so let's remain as is while we do not have strong reason for either of the options.